book/docs/classes/OpusPool/Methods/buildStakeTransaction.md

```diff
@@ -0,0 +1,30 @@
+### buildStakeTransaction
+
+▸ **buildStakeTransaction**(`params`): `Promise`\<[`StakeTransactionData`](../../../interfaces/StakeTransactionData.md)\>
+
+Generates a stake transaction to deposit into the chosen vault
+
+#### Parameters
+
+| Name | Type | Description |
+| :------ | :------ | :------ |
+| `params` | `Object` | Parameters for building the transaction |
+| `params.vault` | \`0x$\{string}\` | A vault address |
+| `params.amount` | `bigint` | An amount of ETH to deposit, denominated in wei |
+| `params.referrer?` | \`0x$\{string}\` | An address of the referrer. Optional |
```

Comment:
The same here: need to expand referrer description so users of SDK will understand importance of setting this, probably need to tell that it should be optional only in testing, and production should have it set.